### PR TITLE
Add "Mark as read" notification action

### PR DESCRIPTION
**Pull request by luk1337**
# AniTrend Pull Request

Thank you for contributing! Please take a moment to review our [**contributing guidelines**](https://github.com/AniTrend/anitrend-app/blob/master/CONTRIBUTING.md)
to make the process easy and effective for everyone involved.

**Please open an issue** before embarking on any significant pull request, especially those that
add a new library or change existing tests, otherwise you risk spending a lot of time working
on something that might not end up being merged into the project.

Before opening a pull request, please ensure:

- [x] You have followed our [**contributing guidelines**](https://github.com/AniTrend/anitrend-app/blob/master/CONTRIBUTING.md)
- [x] double-check your branch is based on `develop` and targets `develop` 
- [ ] Pull request has tests
- [x] Code is well-commented, linted and follows project conventions
- [x] Documentation is updated (if necessary)
- [x] Description explains the issue/use-case resolved
- [x] I did not commit files that are excluded in the .gitignore file (Happens if you stage files with Android Studio)

## Description
This adds a "Mark as read" notification action.
Incidentally, this also colors the notification icon when displayed from the notification shade as otherwise notification actions are grey and look like they are disabled.

## Motivation and Context
Having notification clear on dismiss as opt-in always felt weird to me. This is more natural IMO.

## How Has This Been Tested?
Receive notification, press "Mark as read".

## Screenshots (if appropriate):
ADB Screenshot from 2021-05-17 11-34-35

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

**IMPORTANT**: By submitting a patch, you agree to allow the project
owners to license your work under the terms of the [MIT License](https://github.com/AniTrend/anitrend-app/blob/master/LICENSE.md).

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -474,6 +474,8 @@
     <string name="alerter_notification_title">Notifications Available</string>
     <string name="alerter_notification_text">You have one or more unread notifications!</string>
 
+    <string name="alerter_notification_action_clear">Mark all as read</string>
```
Comment (luk1337) [inline app/src/main/res/values/strings.xml:477]: could also just be "Mark as read"

Comment (taichikuji) [reply app/src/main/res/values/strings.xml:477]: I think it'd be better as "Mark as read".

Thoughts? @wax911

Comment (luk1337) [reply app/src/main/res/values/strings.xml]: I force pushed it to "Mark as read". I too think it's better that way.

Comment (wax911) [reply app/src/main/res/values/strings.xml]: Agreed thanks!

**Review by wax911 — APPROVED** [review]

**Comment by wax911** [comment]
@luk1337 Does this also remove the swipe to dismiss action?

**Comment by luk1337** [comment]
Yes.

**Comment by luk1337** [comment]
If you want I can restore it.

**Comment by wax911** [comment]
> If you want I can restore it.

Yes please, some users mentioned that they may swipe the notification away, expecting it to behave as being marked as read

**Comment by youssefhabri** [comment]
I think the swipe to dismiss notification, but not marking them as read in AniList is a useful feature.

I personally prefer to check my notifications and deal with them mostly on my laptop, so I usually dismiss them so they don't show up again, but I don't want them marked as read in AniList.

So, that option is nice to have. And coupled with the Mark as read, it gives you both options to do what you prefer.

**Comment by luk1337** [comment]
> > If you want I can restore it.
> 
> Yes please, some users mentioned that they may swipe the notification away, expecting it to behave as being marked as read

Done.

File: app/src/main/java/com/mxt/anitrend/util/NotificationUtil.kt [hunk 2]
```diff
@@ -46,13 +47,15 @@ class NotificationUtil(
...
+        val intent = Intent(this.context, ClearNotifications::class.java)
+        intent.putExtra(KeyUtil.NOTIFICATION_ID, defaultNotificationId)
+        intent.putExtra(KeyUtil.NOTIFICATION_ACTION, action)
         return PendingIntent.getBroadcast(
-            this.context,
-            0,
-            onDismissIntent,
-            0
+            context,
+            action.hashCode(),
```
Comment (luk1337) [inline app/src/main/java/com/mxt/anitrend/util/NotificationUtil.kt:56]: With request code hardcoded as 0 I would always receive dismiss action in broadcast receiver thus I'm passing the hashcode here.

Comment (wax911) [reply app/src/main/java/com/mxt/anitrend/util/NotificationUtil.kt:56]: That should be fine 💯

**Review by wax911 — APPROVED** [review]

**Comment by taichikuji** [comment]
Looks good!

**Review by taichikuji — APPROVED** [review]

**Pull request merged**